sexpr/sExpr.ml

```diff
@@ -17,18 +17,30 @@ type t =
   | String of string
   | WeirdString of string * string
 
-open Stdext.Xstringext
-
 let unescape_buf buf s =
   let aux esc = function
       '\\' when not esc -> true
     | c -> Buffer.add_char buf c; false in
-  if String.fold_left aux false s then
+  if Astring.String.fold_left aux false s then
     Buffer.add_char buf '\\'
 
+(* XXX: This escapes "'c'" and "\'c\'" to "\\'c\\'".
+ * They are both unescaped as "'c'". They have been ported
+ * to make sure that this corner case is left unchanged.
+ * It is worth investigating the use of 
+ * - Astring.String.Ascii.escape_string
+ * - Astring.String.Ascii.unescape
+ * that have guaranteed invariants and optimised performances *)
 let escape s =
-  String.map_unlikely s
-    (function '\\' -> Some "\\\\" | '"'  -> Some "\\\"" | '\'' -> Some "\\\'" |  _    -> None)
+  let open Astring in
+  String.fold_left
+    (fun acc c ->
+       acc ^
```

Comment:
I am a bit worried about using string concatenation here. Performance could be a problem. I would suggest to use `Buffer`.

```
let str = Buffer.create (String.length s + 10) in
String.iter 
( function
| ... -> Buffer.add_string str "\\\\"
..
)
; Buffer.contents str
```

Reply:
I agree, I will modify it. Thanks